lib/api/CBoostedTreeInferenceModelBuilder.cc

```diff
@@ -140,12 +140,20 @@ CRegressionInferenceModelBuilder::CRegressionInferenceModelBuilder(const TStrVec
 void CRegressionInferenceModelBuilder::addClassificationWeights(TDoubleVec /*weights*/) {
 }
 
+void CRegressionInferenceModelBuilder::addLossFunction(const maths::CBoostedTree::TLossFunctionUPtr& lossFunction) {
+    m_LossType = lossFunction->type();
+}
+
 void CRegressionInferenceModelBuilder::setTargetType() {
     this->definition().trainedModel()->targetType(CTrainedModel::ETargetType::E_Regression);
 }
 
 void CRegressionInferenceModelBuilder::setAggregateOutput(CEnsemble* ensemble) const {
-    ensemble->aggregateOutput(std::make_unique<CWeightedSum>(ensemble->size(), 1.0));
+    if (m_LossType == TLossType::E_MsleRegression) {
+        ensemble->aggregateOutput(std::make_unique<CExponent>(ensemble->size(), 1.0));
+    } else {
+        ensemble->aggregateOutput(std::make_unique<CWeightedSum>(ensemble->size(), 1.0));
+    }
```

Comment:
My inclination would be to make this a switch: it avoids us forgetting this in future if we add a new regression type.

Reply:
I added a switch and `LOG_ERROR` if we unexpectedly encounter a classification function.

Reply:
Thanks!